File: spl-net/src/main/java/bgu/spl171/net/impl/TFTP/Commands/Responses/Ack.java
package bgu.spl171.net.impl.TFTP.Commands.Responses;

public class Ack implements Response {
    short blockId;

    public Ack(short blockId) {
        this.blockId = blockId;
    }
    public Ack() {
        this.blockId = (short)0;
    }

    public short getBlockId() {
        return blockId;
    }

    @Override
    public short getOpCode() {
        return OpCode.Ack.getValue();
    }

    @Override
    public Response execute() {
        return null;
    }
}

File: java to c++/Responses/Ack.cpp
#include "Ack.h"

namespace bgu
{
	namespace spl171
	{
		namespace net
		{
			namespace impl
			{
				namespace TFTP
				{
					namespace Commands
					{
						namespace Responses
						{

							Ack::Ack(short blockId)
							{
								this->blockId = blockId;
							}

							Ack::Ack()
							{
								this->blockId = static_cast<short>(0);
							}

							short Ack::getBlockId()
							{
								return blockId;
							}

							short Ack::getOpCode()
							{
								return OpCode::Ack::getValue();
							}
						}
					}
				}
			}
		}
	}
}
